xls/passes/rematerialization_pass.cc

```diff
@@ -246,6 +246,10 @@ struct RematOpportunity {
 };
 
 // TODO(zihao): A proposed replacement for `RematOpportunity`. It talks about "edge".
+// TODO(zihao): further refactor this, edges may share the same src.
+//    for example, supposing that dst1 and dst2 are in the same stage, 
+//    "src -> dst1" and "src -> dst2" are cross-stage data dependence stage,
+//    should consider the two edges as a single opportunity.
```

Comment:
I think it's fine to consider them as different opportunities, since adding nodes to the dst1 stage may have different effects on the critical path than adding them to the dst2 stage.